framework/test/unit/modules/interoperability/base_cross_chain_update_command.spec.ts

```diff
@@ -792,85 +796,149 @@ describe('BaseCrossChainUpdateCommand', () => {
 			await expect(
 				command['beforeCrossChainMessagesExecution'](executeContext, true),
 			).resolves.toEqual([[], false]);
+
+			expect(command['verifyRoutingRules']).toHaveBeenCalledWith(
+				ccm,
+				executeContext.params,
+				executeContext.chainID,
+				true,
+			); // ***
+
+			// let's make sure it indeed throws expected error
+			expect(() => {
+				command['verifyRoutingRules'](ccm, executeContext.params, executeContext.chainID, true);
+			}).toThrow('blah');
+
 			expect(internalMethod.terminateChainInternal).toHaveBeenCalledWith(
 				expect.anything(),
 				params.sendingChainID,
 			);
 
-			const ccmID = getIDFromCCMBytes(codec.encode(ccmSchema, ccm));
-			const ccmEventQueue = executeContext.eventQueue.getChildQueue(
-				Buffer.concat([EVENT_TOPIC_CCM_EXECUTION, ccmID]),
-			);
 			expect(command['events'].get(CcmProcessedEvent).log).toHaveBeenCalledWith(
-				{ ...executeContext, eventQueue: ccmEventQueue },
+				expect.anything(),
 				executeContext.params.sendingChainID,
 				ccm.receivingChainID,
 				{
 					ccm,
 					result: CCMProcessedResult.DISCARDED,
-					code: CCMProcessedCode.INVALID_CCM_ROUTING_EXCEPTION,
+					code: CCMProcessedCode.INVALID_CCM_ROUTING_EXCEPTION, // ***
 				},
 			);
+			expect(executeContext.eventQueue['_defaultTopics'][0]).toEqual(
+				Buffer.concat([EVENT_TOPIC_TRANSACTION_EXECUTION, executeContext.transaction.id]),
+			);
 		});
+	});
 
-		// Sending and receiving chains must differ.
-		it('should terminate the chain and add an event when receiving chain is the same as sending chain', async () => {
-			const sendingChainID = chainID;
-			const ccm = {
-				crossChainCommand: CROSS_CHAIN_COMMAND_REGISTRATION,
-				fee: BigInt(0),
-				module: MODULE_NAME_INTEROPERABILITY,
-				nonce: BigInt(1),
-				params: utils.getRandomBytes(10),
-				// must be same as `context.chainID` to pass `!context.chainID.equals(ccm.receivingChainID)` check
-				receivingChainID: chainID,
-				// will fail for `Sending and receiving chains must differ`
-				sendingChainID: chainID,
-				status: CCMStatusCode.OK,
-			};
-
+	describe('verifyRoutingRules', () => {
+		beforeEach(async () => {
 			executeContext = createTransactionContext({
```

Comment:
I see that my statement was not correct. The variable is used, but the value you assign in 845 (current version) is never used because you assign a new value in each test. So either this assignment is redundant or some assignments in the tests.